From 1622001decd3582fabaa0316b3b11a4b1f32a3c8 Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Thu, 11 Dec 2014 09:27:03 +0800
Subject: [PATCH] board: sama5d3xek: change the macros used to generate the
 values for the registers t0pr, t1pr and t2pr in lpddr2_reg_config()

The macro changes of commit: "at91_ddrsdrc: improve the macro definition of the MPDDRC timming parameter register",
broke the lpddr2_reg_config() function in board/sama5d3xek/sama5d3xek.c.

Reported by rwille from the github: linux4sam/at91bootstrap

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
---
 board/sama5d3xek/sama5d3xek.c | 36 ++++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/board/sama5d3xek/sama5d3xek.c b/board/sama5d3xek/sama5d3xek.c
index e6c0ba2..b80e94a 100644
--- a/board/sama5d3xek/sama5d3xek.c
+++ b/board/sama5d3xek/sama5d3xek.c
@@ -238,24 +238,24 @@ static void lpddr2_reg_config(struct ddramc_register *ddramc_config)
 	ddramc_config->rtr = 0x202;
 	ddramc_config->tim_calr = 12;
 
-	ddramc_config->t0pr = (AT91C_DDRC2_TRAS_6
-			| AT91C_DDRC2_TRCD_2
-			| AT91C_DDRC2_TWR_3
-			| AT91C_DDRC2_TRC_8
-			| AT91C_DDRC2_TRP_2
-			| AT91C_DDRC2_TRRD_2
-			| AT91C_DDRC2_TWTR_2
-			| AT91C_DDRC2_TMRD_3);
-
-	ddramc_config->t1pr = (AT91C_DDRC2_TXP_2
-			| AT91C_DDRC2_TXSNR_18
-			| AT91C_DDRC2_TRFC_17);
-
-	ddramc_config->t2pr = (AT91C_DDRC2_TFAW_8
-			| AT91C_DDRC2_TRTP_2
-			| AT91C_DDRC2_TRPA_3
-			| AT91C_DDRC2_TXARDS_1
-			| AT91C_DDRC2_TXARD_1);
+	ddramc_config->t0pr = (AT91C_DDRC2_TRAS_(6)
+			| AT91C_DDRC2_TRCD_(2)
+			| AT91C_DDRC2_TWR_(3)
+			| AT91C_DDRC2_TRC_(8)
+			| AT91C_DDRC2_TRP_(2)
+			| AT91C_DDRC2_TRRD_(2)
+			| AT91C_DDRC2_TWTR_(2)
+			| AT91C_DDRC2_TMRD_(3));
+
+	ddramc_config->t1pr = (AT91C_DDRC2_TXP_(2)
+			| AT91C_DDRC2_TXSNR_(18)
+			| AT91C_DDRC2_TRFC_(17));
+
+	ddramc_config->t2pr = (AT91C_DDRC2_TFAW_(8)
+			| AT91C_DDRC2_TRTP_(2)
+			| AT91C_DDRC2_TRPA_(3)
+			| AT91C_DDRC2_TXARDS_(1)
+			| AT91C_DDRC2_TXARD_(1));
 }
 
 static void lpddr2_init(void)
-- 
1.9.1

